plasmapy/particles/particle_class.py

```diff
@@ -96,6 +97,59 @@ def _category_errmsg(particle, category: str) -> str:
     return errmsg
 
 
+class ParticleJSONDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kwargs):
+        json.JSONDecoder.__init__(self, object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, json_dict):
+        """
+        Returns the appropriate particle object corresponding to the input JSON
+        string. If not a particle, returns a dictionary object. Logic Implementation for
+        plasmapy.particles.from_json method.
+
+        Examples
+        --------
+        >>> import plasmapy
+        >>> input_str = '{"plasmapy_particle": {"type": "Particle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"argument": "Pb"}}}'
+        >>> plasmapy.particles.from_json(input_str)
+        Particle("Pb")
+        >>> input_str = '{"plasmapy_particle": {"type": "CustomParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.12 kg", "charge": "6.2 C"}}}'
+        >>> plasmapy.particles.from_json(input_str)
+        CustomParticle(mass=5.12 kg, charge=6.2 C)
+        >>> input_str = '{"plasmapy_particle": {"type": "DimensionlessParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.2", "charge": "6.3"}}}'
+        >>> plasmapy.particles.from_json(input_str)
+        DimensionlessParticle(mass=5.2, charge=6.3)
```

Comment:
I don't think you need to have examples here, it's not something user's will be playing with and the `pytests` cover its functionality.

Reply:
Makes sense. I have removed the examples from this bit.